builtin/providers/aws/resource_aws_elastic_beanstalk_application_test.go

```diff
@@ -0,0 +1,100 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/elasticbeanstalk"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSBeanstalkApp_basic(t *testing.T) {
+	var app elasticbeanstalk.ApplicationDescription
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckBeanstalkAppDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccBeanstalkAppConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckBeanstalkAppExists("aws_elastic_beanstalk_application.tftest", &app),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckBeanstalkAppDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).elasticbeanstalkconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_elastic_beanstalk_application" {
+			continue
+		}
+
+		// Try to find the VPC
+		DescribeBeanstalkAppOpts := &elasticbeanstalk.DescribeApplicationsInput{
+			ApplicationNames: []*string{aws.String(rs.Primary.ID)},
+		}
+		resp, err := conn.DescribeApplications(DescribeBeanstalkAppOpts)
+		if err == nil {
+			if len(resp.Applications) > 0 {
+				return fmt.Errorf("Elastic Beanstalk Application still exists.")
+			}
+
+			return nil
+		}
+
+		// Verify the error is what we want
+		ec2err, ok := err.(awserr.Error)
+		if !ok {
+			return err
+		}
+		if ec2err.Code() != "InvalidBeanstalkAppID.NotFound" {
+			return err
+		}
```

Comment:
The fact that this error is expected here makes me feel like we should also be catching it up in `Read` too - instead of just a response where `len(Applications) == 0`.